### PR TITLE
[Veeam] Don't interrupt backup syncronization

### PR DESCRIPTION
**Pull request by SadiJr**
### Description

When ACS  is synchronizing the Veeam backups, if one backup fails in this process, all the other backups are skipped and ignored. This behavior is fixed by this PR; if one backup fails in syncronization, only this backup is skipped, and the others continue the process.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?
It was tested in a local lab:

1. I created two VMs, added these VMs to one backup offering, and made two manual backups;
2. I renamed the backup in Veeam of one of then;
3. Before, ACS failed in the syncronization process because the first backup failed to syncronize;
4. Now, the ACS doesn't interrupt the synchronization process of the rest of the backups.

### CODE REVIEW

File: server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java [hunk 1]
```diff
@@ -1096,6 +1096,26 @@ protected void runInContext() {
...
+                        backupProvider.syncBackups(vm, metric);
+                        // Emit a usage event, update usage metric for the VM by the usage server
+                        UsageEventUtils.publishUsageEvent(EventTypes.EVENT_VM_BACKUP_USAGE_METRIC, vm.getAccountId(),
+                                vm.getDataCenterId(), vm.getId(), "Backup-" + vm.getHostName() + "-" + vm.getUuid(),
+                                vm.getBackupOfferingId(), null, metric.getBackupSize(), metric.getDataSize(),
+                                Backup.class.getSimpleName(), vm.getUuid());
+                    }
+                } catch (final Exception e) {
+                    LOG.error(String.format("Failed to sync backup usage metrics and out-of-band backups of VM [%s] due to: [%s].", vm.getUuid(), e.getMessage()), e);
+                }
```
Comment (DaanHoogland) [inline server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java:1115]: for readability the try block can be a separate method.

Comment (SadiJr) [reply server/src/main/java/org/apache/cloudstack/backup/BackupManagerImpl.java]: Done, thanks for the suggestion.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 vmware-70u2

**Review by harikrishna-patnala — APPROVED** [review]
code LGTM

**Review by DaanHoogland — APPROVED** [review]
clgtm, will trust the users to approve with testing

**Comment by DaanHoogland** [comment]
@blueorangutan test rocky8 vmware-70u3

**Comment by harikrishna-patnala** [comment]
@blueorangutan test rocky8 vmware-70u3

**Comment by GutoVeronezi** [comment]
@blueorangutan package

@DaanHoogland @harikrishna-patnala could we run another round of tests? Seems that we had a problem with blueorangutan in the last run.

**Comment by weizhouapache** [comment]
@blueorangutan test rocky8 vmware-70u3

**Comment by harikrishna-patnala** [comment]
@blueorangutan test rocky8 vmware-70u3

**Comment by SadiJr** [comment]
@DaanHoogland @harikrishna-patnala @GutoVeronezi could we run another round of tests? Seems that we had a problem with blueorangutan in the last run.

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

**Comment by harikrishna-patnala** [comment]
@blueorangutan test rocky8 vmware-70u3

**Comment by DaanHoogland** [comment]
insanity:
@blueorangutan test rocky8 vmware-70u3

**Comment by shwstppr** [comment]
Merging based on reviews and smoke test results

**Pull request merged**